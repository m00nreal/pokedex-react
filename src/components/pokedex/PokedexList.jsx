import React from 'react';
import PokeCard from "./PokedexCard";
import {Spinner} from "reactstrap";

const PokedexList = ({ data, loading }) => {


    return (
        <>
            <div className="container">
                <div className="row">
                    {
                        loading ?
                            <div className="col m-4 d-flex justify-content-center align-items-center">
                                <Spinner color="primary"/>
                            </div>
                            :
                            data.map(pokemon => (
                                <div className="col-xs-12 col-sm-6 col-md-4 col-lg-3 mb-4" key={pokemon.id}>
                                    <PokeCard
                                        key={pokemon.id}
                                        {...pokemon}
                                        loading={loading}
                                    />
                                </div>
                            ))
                    }
                </div>
            </div>
        </>
    );
};

export default PokedexList;
