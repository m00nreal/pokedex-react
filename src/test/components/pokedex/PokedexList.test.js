import PokedexList from "../../../components/pokedex/PokedexList";
import {shallow} from "enzyme";
import React from "react";


describe('Tests in <PokedexList/>', function () {

    it('should match snapshot', function () {

        const data = [{
            id: 1,
            image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
            name: 'bulbasaur',
            abilities: ['overgrow, chlorophyll'],
            height: 70,
            weight: 69,
            types: ['grass', 'poison'],
        }]

        const wrapper = shallow(<PokedexList data={data} loading={false}/>)
        expect(wrapper).toMatchSnapshot();
    });

});