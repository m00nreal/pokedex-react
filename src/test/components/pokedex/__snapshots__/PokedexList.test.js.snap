// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PokedexList/> should match snapshot 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6 col-md-4 col-lg-3 mb-4"
        key="1"
      >
        <PokeCard
          abilities={
            Array [
              "overgrow, chlorophyll",
            ]
          }
          height={70}
          id={1}
          image="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
          key="1"
          loading={false}
          name="bulbasaur"
          types={
            Array [
              "grass",
              "poison",
            ]
          }
          weight={69}
        />
      </div>
    </div>
  </div>
</Fragment>
`;
